### PR TITLE
[CDU] Reduced fuel pred delay and average wind change delay

### PR DESCRIPTION
**Pull request by Lucky38i**
## Summary of Changes

Just changed the fuel prediction delays in Init B and changed the method usage of changing average wind. This was due to a number of people saying it was a tad too slow. The fuel prediction delay is now halved however it can still vary depending on the number of waypoints you have and destination distance.

Discord username (if different from GitHub): Lucky38i

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

**Comment by danbw9** [comment]
Category: QA Tester
Name: danbw#7028
Date of testing: 20/11/2020
Version of the sim: 1.10.11.0
PR Tested: #2014 
Object of test: Reduced FUEL PRED Delay + wind change delay

Testing technique: Spawn C&D on two flights one short one and one medium one
Steps you've made to test the Pull Request:

(First Check) Spawn C&D at EDDK for flight to EKCH.
- Input flight plan in full with SID and STAR. 
- put in the ZFW/ZFWCG and BLK FUEL
- wait for INIT B calculation. Expected it to be completed within a few seconds and not 10 seconds. Route below. 

Route: PODI5D PODIP Y867 TOLGI DCT ROSOK Q296 MONAK MONA2D


(Second Check) Spawn C&D at ESSA for flight to LEMG. 
- same as first check, inputting all data. 
- after BLK fuel entry, timed until all calculations done.
- Total of 21 seconds to calculate
- Same flight with similar route took 6 seconds to calculate in justplanes video for comparison

Route: NOSL5G NOSLI N850 ABAMA DCT MAXEL DCT BASUM DCT RASCA DCT
           FAMEN DCT BSN DCT BUZIK DCT RESMI/N0450F350 UN857 DISAK DCT ARKIP
           DCT ARNAV DCT LMG UN857 OSMOB DCT ARVID UT424 PPN UN10 BAN UN857
           MONTO UN864 VULPE VULP2K

Wind change delay:
- first route the calculations were made instantanteous on short route and within a second on the medium route

Issues (If Any): Still too slow. 
Overall Rating: Mostly good
Conclusions of the testing: Better than before @Lucky38i . I just feel it's still too slow in my opinion. Wonder what other IRL A320 pilots think on this - mainly those with honeywell experience. Now my experience is much more with THALES systems and not Honeywell, but anyway I cannot see how it would be so different. I have a JustPlanes A320neo SAS video as reference which I have shown to @Lucky38i . I have left a swisstransfer link to the mp4 I have as a reference here: 
https://www.swisstransfer.com/d/d939b3b4-24ae-4566-8c45-188377ea0efa

**Comment by FlyingCarpet7** [comment]
The 10 and 6 seconds example above I feel would be more like 6 and 4 seconds in real life. Our older Honeywell’s are sure slower then the newer Thales.

**Comment by Lucky38i** [comment]
> The 10 and 6 seconds example above I feel would be more like 6 and 4 seconds in real life. Our older Honeywell’s are sure slower then the newer Thales.

Yep I've added a few more commits that has reduced the delay quite a bit. Can be retested

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/FMC/A32NX_FMCMainDisplay.js [hunk 1]
```diff
@@ -3029,7 +3029,8 @@ class FMCMainDisplay extends BaseAirliners {
      * @returns {number} dynamic delay in ms between 2000ms and 4000ms
      */
     getDelayFuelPred() {
-        return 1714 * this.flightPlanManager.getWaypointsCount() + this.flightPlanManager.getDestination().cumulativeDistanceInFP;
+        const tripDist = this.flightPlanManager.getDestination().cumulativeDistanceInFP;
+        return 225 * this.flightPlanManager.getWaypointsCount() + (tripDist > 0 ? tripDist / 2 : 0);
```
Comment (lukecologne) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/FMC/A32NX_FMCMainDisplay.js:3033]: What is this ternary statement for? Can the cumulative distance get negative?

Comment (Lucky38i) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/FMC/A32NX_FMCMainDisplay.js]: I thought there'd be an exception if I tried to divide by 0 but I tried it out in jsfiddle and it's ok

Comment (lukecologne) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/FMC/A32NX_FMCMainDisplay.js]: You're not dividing by zero, you're dividing 0 by two ;)

**Comment by danbw9** [comment]
Hey @Lucky38i I gave this another test. A lot more reasonable now after the recent commits pushed. Thanks a lot.

**Comment by Lucky38i** [comment]
> Hey @Lucky38i I gave this another test. A lot more reasonable now after the recent commits pushed. Thanks a lot.

Hey @danbw9 thanks for the reply, would you say it needs to be lessened further?

**Comment by Carter633** [comment]
Name: British Avgeek UTube -- Carter633 #1436
Date of testing: Current date DD/MM/YYYY
Version of the sim: 1.10.11.0
PR Tested: #2014 
Object of test: Reduced fuel pred delay/wind change delay in MCDU

Testing technique: Full Flight on VATSIM - EHAM to EGLL
Steps you've made to test the Pull Request:
1. Spawn C&D - generate route using Simbrief and input into MCDU
2. Fuel planning was quick
3. Fuel prediction was quick too as was  wind change delay 
4. Faster than previous build - nice work

Issues (If Any): None
Overall Rating: Good
Conclusions of the testing: Full flight, no issues during any phase and fuel pred delay was probably halved compared to before as was wind change delay!

**Comment by danbw9** [comment]
@Lucky38i As we are dealing with honeywell I cannot say exactly unfortunately , but I understand that honeywell is a lot slower and in my very little experience as well when I flew old A320s. Thales is what I am used to. I have no idea how quick the computation is with the newer honeywell on this NEO in seconds, but this is certainly more realistic than before!

**Comment by Benjozork** [comment]
Alright, now we need an accuracy check from someone familiar with Thales.

**Comment by St54Kevin** [comment]
Category: Accuracy Check
Name: St54Kevin #3583
Date of testing: Current date 23/11/2020
Version of the sim: 1.10.11.0
PR Tested: #2014 
Object of test: INIT B fuel calculation duration

Expected IRL Behavior:
We operate the Thales System. We experience durations from 1~5 seconds.

Current PR behavior:
Spawned C&D at the Gate in FRA. Put in the flight plan and then Block fuel and ZWF/CG in INIT B. Calculation took 5 seconds. I found that very reasonable. I dont know what the OP means with wind calculation. I manually edited the wind component to TL037 but it didnt change anything.

Flight notes (If required):
EDDF-LOWW
CIND1D CINDY Z74 LAMPU DCT NOMBO Y161 RIDAR UZ98 MIQ UL605 BIBAG Q104 AKIMA DCT MASUR MASU2W
TOW: 65.9t

Overall Rating: LGTM
Conclusions of the testing: Duration looks good to me. IRL it really depends heavily on the computer power and complexity of the flight plan, so you cannot establish a dependend system there anyway.

**Comment by Lucky38i** [comment]
> Category: Accuracy Check
> Name: St54Kevin #3583
> Date of testing: Current date 23/11/2020
> Spawned C&D at the Gate in FRA. Put in the flight plan and then Block fuel and ZWF/CG in INIT B. Calculation took 5 seconds. I found that very reasonable. I dont know what the OP means with wind calculation. I manually edited the wind component to TL037 but it didnt change anything.

Thanks for the review @St54Kevin, What I meant was previously when you entered a wind component the method called was the same as the fuel prediction. I changed the method to a much short method call, shorter than the fuel prediction delay.

**Pull request merged**